fpdf/drawing.py

```diff
@@ -0,0 +1,1023 @@
+        try:
+            return self._get_style(StrokeJoinStyle)
+        except KeyError:
+            return None
+
+    @stroke_join_style.setter
+    def stroke_join_style(self, style):
+        self._set_style(StrokeJoinStyle.coerce(style))
+
+    @property
+    def stroke_miter_limit(self):
+        try:
+            return self._get_style(StrokeMiterLimit).limit
+        except KeyError:
+            return None
+
+    @stroke_miter_limit.setter
+    def stroke_miter_limit(self, limit):
+        self._set_style(StrokeMiterLimit(limit))
+
+    @property
+    def transform(self):
+        try:
+            return self._get_style(Transform)
+        except KeyError:
+            return None
+
+    @transform.setter
+    def transform(self, tf):
+        self._set_style(tf)
+
+    @property
+    def clipping_path(self):
+        try:
+            return self._get_style(ClippingPath)
+        except KeyError:
+            return None
+
+    @clipping_path.setter
+    def clipping_path(self, path):
+        self._set_style(path)
+
+
+class PaintedPath(Stylable):
+    def __init__(self, paint_style, x=0, y=0):
+        self._root_graphics_context = GraphicsContext()
+        self._graphics_context = self._root_graphics_context
+
+        self._paint_style = PaintStyle.coerce(paint_style)
+        self._auto_close = self._paint_style in {
+            PaintStyle.close_stroke,
+            PaintStyle.close_stroke_fill_nonzero,
+            PaintStyle.close_stroke_fill_evenodd,
+        }
+
+        # self._auto_close = False
+
+        self._closed = True
+        self._close_context = self._graphics_context
+        self.move_to(x, y)
+
+    @contextmanager
+    def _new_graphics_context(self, _attach=True):
+        old_graphics_context = self._graphics_context
+        new_graphics_context = GraphicsContext()
+        self._graphics_context = new_graphics_context
+        try:
+            yield new_graphics_context
+            if _attach:
+                old_graphics_context.add_item(new_graphics_context)
+        finally:
+            self._graphics_context = old_graphics_context
+
+    @contextmanager
+    def transform_group(self, transform):
+        """
+        Apply the provided transforms to all points added within this context.
+        """
+        with self._new_graphics_context() as ctxt:
+            ctxt.transform = transform
+            yield self
+
+    def _push_drawing_sequence(self, item):
+        if self._starter_move is not None:
+            self._closed = False
+
+            self._graphics_context.add_item(self._starter_move)
+            self._close_context = self._graphics_context
+            self._starter_move = None
+
+        self._graphics_context.add_item(item)
+
+    def rectangle(self, x, y, w, h):
+        """
+        Append a rectangle as a closed subpath to the current path.
+
+        The rectangle has all of the same constraints that regular subpaths do.
+        """
+        self._close()
```

Comment:
Seems like all public "drawing" methods of this class call `self._close()` as a starting point.
Could you please a short comment at the top of this class to explain why?

Reply:
Hopefully my comment above mostly explained this, but only the drawing methods that create a new subpath within a single painting context call `self._close()`. This is something that I can document better in the code as well.